common/dynamicconfig/constants.go

```diff
@@ -2045,6 +2045,11 @@ the user has not specified an explicit RetryPolicy`,
 		`DefaultWorkflowRetryPolicy represents the out-of-box retry policy for unset fields
 where the user has set an explicit RetryPolicy, but not specified all the fields`,
 	)
+	AllowResetWithPendingChildren = NewNamespaceBoolSetting(
+		"history.allowResetWithPendingChildren",
```

Comment:
we probably want to be more specific here so that we can enable it for only phase 1 of the project.